### DISCUSSION TITLE
Using OSMD with XML string as input, instead of XML file?

### DISCUSSION THREAD

**Discussion by pieguyatx** [opensheetmusicdisplay/opensheetmusicdisplay#1027] [Q&A] [answered]
Hello! Thanks for creating this. Your repo is really interesting, and I'm interested in using it to let students create sheet music for [algebraic sonifications](https://www.pioslabs.com/chordinates/).  

Is there a way to use OSMD on an input string, instead of loading a whole XML or MusicXML file?  I was trying to figure out how to load a "score-partwise" code block into OSMD (auto-generated in the browser), so users don't have have to download any files, but no luck.  I was just going off your raw JS example, so I'm not sure if that's possible with that option.

**Comment by sschmidTU** [comment, marked as answer]
Yes, you can just load a string (that is the content of a MusicXML file) in osmd.load() instead of a file.
The string has to be a valid MusicXML document (content) though.

**Reply by pieguyatx** [reply]
thanks!
